### PR TITLE
Configure document's child restrictions

### LINKED ISSUE #272: Configure PHPCR ODM child restrictions
We should implement https://github.com/doctrine/phpcr-odm/pull/706 in 2.0 of this bundle. The tree browser bundle can use this information to restrict children.

### PR DESCRIPTION
**Pull request by wouterj**
/fixes #272

### CODE REVIEW

File: src/Resources/config/doctrine-phpcr/ImagineBlock.phpcr.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
         name="Symfony\Cmf\Bundle\BlockBundle\Doctrine\Phpcr\ImagineBlock"
         referenceable="true"
         translator="attribute"
+        is-leaf="true"
```
Comment (dbu) [inline src/Resources/config/doctrine-phpcr/ImagineBlock.phpcr.xml]: are you sure this works? the image is a jcr:file node which afaik is translated to be a child document

File: src/Resources/config/doctrine-phpcr/SlideshowBlock.phpcr.xml [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
         <field name="title" type="string" translated="true" nullable="true"/>
 
+        <child-class name="Symfony\Cmf\Bundle\BlockBundle\Doctrine\Phpcr\ImagineBlock"/>
```
Comment (dbu) [inline src/Resources/config/doctrine-phpcr/SlideshowBlock.phpcr.xml]: is this really a requirement? or could a slideshow slide any kind of blocks? what does our template expect? if it handles any kind of block, we should handle this like the ContainerBlock

File: src/Resources/config/doctrine-phpcr/ImagineBlock.phpcr.xml (hunk 1 above)
Comment (wouterj) [inline src/Resources/config/doctrine-phpcr/ImagineBlock.phpcr.xml]: Hmm, I don't know and I can't find any tests of this block. However, I think we have to remove the `ImagineBlock` from the 2.0 release as we abandoned the MediaBundle (this block relies on the media bundle). Wdyt?

File: src/Resources/config/doctrine-phpcr/SlideshowBlock.phpcr.xml (hunk 2 above)
Comment (wouterj) [inline src/Resources/config/doctrine-phpcr/SlideshowBlock.phpcr.xml]: It indeed behaves like the ContainerBlock. I've updated this PR to reflect this. (I took the original child restrictions from the valid_children config of the CMF sandbox)

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -66,3 +66,14 @@
 
  * The `ImagineBlock` was removed from this bundle as the CmfMediaBundle is
    abandoned. Implement the block in your own application if you were using it.
+
+### Doctrine PHPCR ODM
+
+ * It is no longer possible to add a child to the `ActionBlock`,
+   `ImagineBlock`, `MenuBlock`, `ReferenceBlock`, `RssBlock`, `SimpleBlock` and
```
Comment (dbu) [inline UPGRADE-2.0.md]: remove ImagineBlock, you just removed that (correctly!) in #274

**Comment by wouterj** [comment]
@dbu thanks! Fyi, I didn't miss your comment, I included it in 28d8262ad8315155a48e5fbf5d9ddb24b1dbf9e4 to avoid needless Travis builds.

**Pull request merged**